### PR TITLE
Add option for pull_theme and install latest shopify-cli 2.x

### PR DESCRIPTION
**Pull request by stufreen**
### Summary

* Adds an optional param for `pull_theme` which, if provided, will pull the JSON templates and settings values for a specified theme
* Update Dockerfile to reinstall `shopify-cli` gem. The image has an older version of shopify-cli (2.0.1) that is missing key `theme pull` options

### Why I'm adding it

On the Theme Design team we need to track lighthouse scores for realistic storefronts, e.g. those with image heroes and many sections. We also may need to test different configurations (various combinations of sections, menu styles, etc.) However, the lighthouse action currently only tests against the default settings of a theme.

### Approach taken

1. Uninstalls the existing `shopify-cli` gem and reinstalls the latest version. I found this was preferable to rebuilding and re-hosting the docker image with a new `shopify-cli` install but I'm open to it if there's someone available to assist.
2. Checks for a `pull_theme` option passed to the github action. This should be a theme name or theme ID, same as `shopify theme pull --theme`.
3. If `pull_theme` exists it triggers an extra step in the script to pull the theme settings.

### Testing instructions

Add a CI job to a theme pointing to the latest SHA in this PR:

```
name: CI
on: [push]
jobs:
  lhci:
    name: Lighthouse
    runs-on: ubuntu-latest
    steps:
      - uses: actions/checkout@v2
      - name: Lighthouse
        uses: shopify/lighthouse-ci-action@f63f6c39a4cb6fb40169194dcf51eac62354651c
        with:
          store: ${{ secrets.SHOP_STORE_OS2 }}
          password: ${{ secrets.SHOP_PASSWORD_OS2 }}
          access_token: ${{ secrets.SHOP_ACCESS_TOKEN }}
          pull_theme: ${{ secrets.SHOP_PULL_THEME }}
          product_handle: the-organic-cotton-cutaway-tank
          collection_handle: all
          lhci_github_app_token: ${{ secrets.LHCI_GITHUB_TOKEN }}
```

Run the lighthouse-ci action, e.g. by posting a new PR.

Check the lighthouse reports generated by the action. Confirm that the settings from the theme specified were pulled successfully.

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -1,3 +1,5 @@
 FROM cpclermont/lighthouse-ci-action:1.0.0
+RUN gem uninstall shopify-cli
+RUN gem install shopify-cli -N -n /usr/local/bin
```
Comment (stufreen) [inline Dockerfile:3]: It seems that more recent versions of `shopify-cli` gem 2.x are installed to a different directory by default and require some changes to the PATH. Installing the executable to `/usr/local/bin` gets around this.

**Comment by stufreen** [comment]
In this screenshot you can see it working where the Image Banner section has its settings pulled from another theme.

Screenshot 2023-02-24 at 4 44 38 PM

File: action.yml [hunk 2]
```diff
@@ -26,6 +26,9 @@ inputs:
     description: 'The root folder for the theme files that will be uploaded (default: ".")'
     required: false
     default: '.'
+  pull_theme:
+    description: 'If provided, the settings and json templates will be pulled into the test theme.'
```
Comment (charlespwd) [inline action.yml:30]: The description made me feel like this was a boolean, can we be more explicit about the type of the expected value?

Something like:

```
'The ID of a theme from which the settings and JSON templates should be pulled. If provided, those settings will be pulled into the development theme.'
```

Also, can this be added to the README as well?

File: entrypoint.sh [hunk 3]
```diff
@@ -159,6 +160,12 @@ theme_root="${THEME_ROOT:-.}"
 log "Will run Lighthouse CI on $host"
 
 step "Creating development theme"
+
+if [[ -n "${SHOP_PULL_THEME+x}" ]]; then
+  log "Pulling settings from theme $SHOP_PULL_THEME"
+  shopify theme pull --theme ${SHOP_PULL_THEME} --only templates/*.json --only config/settings_data.json
+fi
```
[the remark this answers is not in the document]
Comment (stufreen) [reply entrypoint.sh:167]: Are you suggesting an extra step in the CI process to do a `theme pull` or equivalent? e.g.
```
  lhci:
    name: Lighthouse
    runs-on: ubuntu-latest
    steps:
      - uses: actions/checkout@v2
      - uses: shopify/theme-pull
      - uses: shopify/lighthouse-ci-action
```
That could work, although I feel like adding a new github action for theme pull would be more duplicative than adding the functionality here. We could probably do it with `git` manually but it seems a little less reliable than shopify cli.

Alternatively, if the suggestion is to store the setting values in git, I don't think that works. Most themes don't store any content (images, etc) in the default settings and these really affect the lighthouse scores. Live stores generally have way more images and sections than fresh installs.

For example, these two stores are the same theme:
Stock dawn: https://apod2mu9x7lc1q0n-20706295830.shopifypreview.com
Dawn demo store: https://theme-dawn-demo.myshopify.com/

[the remark this answers is not in the document]
Comment (Poitrin) [reply entrypoint.sh:167]: >  if the suggestion is to store the setting values in git, I don't think that works. Most themes don't store any content (images, etc) in the default settings and these really affect the lighthouse scores

I guess I was missing this, thanks for the clarification!

**Comment by stufreen** [comment]
@charlespwd @Poitrin Any more requests/thoughts on this one?

**Review by Poitrin — APPROVED** [review]
LGTM!

**Pull request merged**